### PR TITLE
add new `revertibleRandom` function

### PR DESCRIPTION
**Pull request by tarakby**
Closes https://github.com/onflow/cadence/issues/2869

Please also review the doc changes https://github.com/onflow/docs/pull/423 
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation (https://github.com/onflow/docs/pull/423)
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/common/computationkind.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ const (
 	//
 	ComputationKindSTDLIBPanic
 	ComputationKindSTDLIBAssert
-	ComputationKindSTDLIBUnsafeRandom
+	ComputationKindSTDLIBRevertibleRandom
```
Comment (tarakby) [inline runtime/common/computationkind.go:136]: Not sure if I should keep `ComputationKindSTDLIBUnsafeRandom` here, knowing that `unsafeRandom` will be deprecated.

File: runtime/common/computationkind_string.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func _() {
 	_ = x[ComputationKindEncodeValue-1080]
 	_ = x[ComputationKindSTDLIBPanic-1100]
 	_ = x[ComputationKindSTDLIBAssert-1101]
-	_ = x[ComputationKindSTDLIBUnsafeRandom-1102]
+	_ = x[ComputationKindSTDLIBRevertibleRandom-1102]
```
Comment (tarakby) [inline runtime/common/computationkind_string.go:27]: file updated through `go generate`

File: runtime/stdlib/builtin.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ package stdlib
 
 type StandardLibraryHandler interface {
 	Logger
-	UnsafeRandomGenerator
+	RandomGenerator
```
Comment (tarakby) [inline runtime/stdlib/builtin.go:23]: renamed the generator name

**Review by dsainati1 — APPROVED** [review]

File: runtime/common/computationkind.go (hunk 1 above)
Comment (turbolent) [reply runtime/common/computationkind.go:136]: I think that's OK, on the FVM side we can probably just use the weights for the old kind for the new one, right @onflow/flow-execution-and-verification ?

File: runtime/stdlib/random.go [hunk 4]
```diff
@@ -26,26 +26,56 @@ import (
...
+					return binary.LittleEndian.Uint64(buffer[:])
+				},
+			)
+		},
+	)
+}
+
+// `unsafeRandom` related constants and functions will be deleted
+// when the function is deprecated
+const unsafeRandomFunctionDocString = revertibleRandomFunctionDocString
```
Comment (turbolent) [inline runtime/stdlib/random.go:74]: Maybe add copy of the docstring for `revertibleRandom` here, but also prefix it with a deprecation warning. That way the language server will show developers a deprecation warning. Finally, in https://github.com/onflow/cadence-tools/blob/9f82dd687af84e8b6d2ec21c52d80a2b42146009/lint/deprecated_member_analyzer.go#L30 we can then suggest a known replacement

**Review by turbolent — APPROVED** [review]
Nice!

File: runtime/stdlib/random.go [hunk 5]
```diff
@@ -26,26 +26,64 @@ import (
...
+				},
+			)
+		},
+	)
+}
+
+// `unsafeRandom` related constants and functions will be deleted
+// when the function is deprecated
+const unsafeRandomFunctionDocString = `
+Warning: this function will be deprecated, use revertibleRandom function instead.
```
Comment (turbolent) [inline runtime/stdlib/random.go:75]: Use "deprecated" instead, as it is detected by the tooling (https://github.com/onflow/cadence-tools/blob/9f82dd687af84e8b6d2ec21c52d80a2b42146009/lint/deprecated_member_analyzer.go#L58) and annotates uses of the function with a warning:

```suggestion
Deprecated: Use revertibleRandom instead.
```

**Pull request merged**